tools/linker/RemoveRejectedTypesStep.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Collections.Generic;
+
+using Mono.Cecil;
+using Mono.Linker;
+using Mono.Tuner;
+using MonoTouch.Tuner;
+using Xamarin.Tuner;
+
+namespace Xamarin.Linker {
+
+	public class RemoveRejectedTypesStep : RemoveCodeBase {
+
+		public RemoveRejectedTypesStep ()
+		{
+		}
+
+		public DerivedLinkContext Context {
+			get { return context as DerivedLinkContext; }
+		}
+
+		public override SubStepTargets Targets {
+			get { return SubStepTargets.Assembly; }
+		}
+
+		protected override string Name { get; } = " Removing Rejected Type";
+		protected override int ErrorCode { get; } = 2660;
+
+		public List<string> ReferencesToBeRemoved = new List<string> () {
+			// order is important for nested types
+			"UIKit.UIWebView/_UIWebViewDelegate",
+			"UIKit.UIWebView/UIWebViewAppearance",
+			"UIKit.IUIWebViewDelegate",
+			"UIKit.UIWebView",
+			"UIKit.UIWebViewDelegate",
+			"UIKit.UIWebViewDelegate_Extensions",
+			"UIKit.UIWebViewDelegateWrapper",
+			"UIKit.UIWebViewNavigationType",
+		};
+
+		protected override void Process (AssemblyDefinition assembly)
+		{
+			// this will give us a working `get_nse`
+			base.Process (assembly);
+
+			var updated = false;
+			if (Profile.IsProductAssembly (assembly)) {
+				updated = true;
+				foreach (var remove in ReferencesToBeRemoved) {
+					var type = assembly.MainModule.GetType (remove);
+					if (type == null)
+						continue;
+					type.Name = type.Name.Replace ("UI", "Deprecated");
+					Context.AddLinkedAwayType (type);
+					// remove [Register], [Protocol] and [ProtocolMembers] attributes
+					type.CustomAttributes.Clear ();
+					if (!type.IsInterface) {
+						// remove IL and custom attributes (e.g. [Export])
+						ProcessMethods (type);
+					}
+				}
+			} else if (Profile.IsSdkAssembly (assembly)) {
+				// note: we know the BCL does not include references to UIWebView
```

Comment:
This might not be correct when something new is added to the list of types to remove. Not sure what to do here though. One solution would be to (again) make `ReferencesToBeRemoved` a hash table with a boolean lookup value that states whether the type can be found in SDK assemblies or not.

Reply:
This situation is very uncommon - it's the first time, after a dozen years, that a required removal affect our public API. All other times we only had to change our internals to avoid the _rejected_ API.

I don't think we need the extra complexity. Worse case we can revisit the code if/when that situation change but my guess is that, because of naming convention, the public BCL types are unlikely to be a trigger.